snowflake/src/main/java/org/apache/iceberg/snowflake/SnowflakeCatalog.java

```
@@ -0,0 +1,204 @@
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.exceptions.NoSuchNamespaceException;
+import org.apache.iceberg.hadoop.Configurable;
+import org.apache.iceberg.io.FileIO;
+import org.apache.iceberg.jdbc.JdbcClientPool;
+import org.apache.iceberg.jdbc.UncheckedInterruptedException;
+import org.apache.iceberg.jdbc.UncheckedSQLException;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+import org.apache.iceberg.snowflake.entities.SnowflakeSchema;
+import org.apache.iceberg.snowflake.entities.SnowflakeTable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SnowflakeCatalog extends BaseMetastoreCatalog
+    implements Closeable, SupportsNamespaces, Configurable<Object> {
+
+  private static final Logger LOG = LoggerFactory.getLogger(SnowflakeCatalog.class);
+
+  private Object conf;
+  private String catalogName = SnowflakeResources.DEFAULT_CATALOG_NAME;
+  private Map<String, String> catalogProperties = null;
+  private FileIO fileIO;
+  private QueryFactory queryFactory;
+
+  public SnowflakeCatalog() {}
+
+  public void setQueryFactory(QueryFactory factory) {
+    queryFactory = factory;
+  }
+
+  @Override
+  public List<TableIdentifier> listTables(Namespace namespace) {
+    Preconditions.checkArgument(
+        namespace.length() <= SnowflakeResources.MAX_NAMESPACE_DEPTH,
+        "Snowflake doesn't supports more than 2 levels of namespace");
+
+    List<SnowflakeTable> sfTables = queryFactory.listIcebergTables(namespace);
+
+    return sfTables.stream()
+        .map(
+            table ->
+                TableIdentifier.of(table.getDatabase(), table.getSchemaName(), table.getName()))
+        .collect(Collectors.toList());
+  }
+
+  @Override
+  public boolean dropTable(TableIdentifier identifier, boolean purge) {
```

We should probably throw UnsupportedOperationException or whatever is idiomatic within Iceberg instead of silently ignoring the ones we're not implementing.

---

Done